web/pdf_viewer.js

```diff
@@ -2122,54 +2132,52 @@ class PDFViewer {
    * @property {number} [drawingDelay]
    * @property {number} [scaleFactor]
    * @property {number} [steps]
+   * @property {Array} [origin] x and y coordinates of the scale
+   *                            transformation origin
    */
 
   /**
-   * Increase the current zoom level one, or more, times.
+   * Changes the current zoom level by the specified amount.
    * @param {ChangeScaleOptions} [options]
    */
-  increaseScale({ drawingDelay, scaleFactor, steps } = {}) {
+  updateScale({ drawingDelay, scaleFactor = null, steps = null, origin }) {
+    if (steps === null && scaleFactor === null) {
+      throw new Error(
+        "Invalid updateScale options: either `scale` or `scaleFactor` must be provided."
+      );
+    }
     if (!this.pdfDocument) {
       return;
     }
     let newScale = this._currentScale;
-    if (scaleFactor > 1) {
+    if (scaleFactor > 0 && scaleFactor !== 1) {
       newScale = Math.round(newScale * scaleFactor * 100) / 100;
-    } else {
-      steps ??= 1;
+    } else if (steps) {
+      const delta = steps > 0 ? DEFAULT_SCALE_DELTA : 1 / DEFAULT_SCALE_DELTA;
+      const round = steps > 0 ? Math.ceil : Math.floor;
+      steps = Math.abs(steps);
       do {
-        newScale =
-          Math.ceil((newScale * DEFAULT_SCALE_DELTA).toFixed(2) * 10) / 10;
-      } while (--steps > 0 && newScale < MAX_SCALE);
+        newScale = round((newScale * delta).toFixed(2) * 10) / 10;
+      } while (--steps > 0 && newScale < MAX_SCALE && newScale > MIN_SCALE);
```

Comment:
Given the clamping just below, do we still need this "full" condition here or could it be simplified as follows?
(Obviously it'd lead to a tiny bit of unnecessary computation at the ends of the [MIN_SCALE, MAX_SCALE] range, but I don't know how much that really matters.)
```suggestion
      } while (--steps > 0);
```

Reply:
I assume this is not called in a hot path, and that usually the number of steps is low anyway, so yes this should be fine.

Reply:
> I assume this is not called in a hot path, and that usually the number of steps is low anyway,

Correct.